asllib/Typing.ml

```diff
@@ -542,41 +565,42 @@ module Annotate (C : ANNOTATE_CONFIG) : S = struct
     | _ -> conflict loc [ T_Exception [] ] t_struct
   (* End *)
 
-  (* Begin StorageIsPure *)
-  let storage_is_pure ~loc (env : env) s =
-    (* Definition DDYW:
-       Any expression consisting solely of an immutable storage element or a
-       literal value is a statically evaluable expression.
-    *)
-    match IMap.find_opt s env.local.storage_types with
-    | Some (_, (LDK_Constant | LDK_Let)) -> true
-    | Some (_, LDK_Var) -> false
-    | None -> (
-        match IMap.find_opt s env.global.storage_types with
-        | Some (_, (GDK_Constant | GDK_Config | GDK_Let)) -> true
-        | Some (_, GDK_Var) -> false
-        | None ->
-            if
-              IMap.mem s env.global.subprograms
-              || IMap.mem s env.global.declared_types
-            then false
-            else undefined_identifier loc s)
-  (* End *)
-
-  (* Begin IsStaticallyEvaluable *)
-  let is_statically_evaluable ~loc env e =
-    let use_set = use_e e ISet.empty in
-    ISet.for_all (storage_is_pure ~loc env) use_set
-    |: TypingRule.IsStaticallyEvaluable
-  (* End *)
-
   (* Begin CheckStaticallyEvaluable *)
-  let check_statically_evaluable (env : env) e () =
-    if is_statically_evaluable ~loc:e env e then
+  let check_statically_evaluable e ses () =
```

Comment:
Can we keep the convention that annotated types used just for errror report be named something like `loc` or `error_annot`?